### PR TITLE
[Merged by Bors] - chore (Algebra.PUnitInstances): split file into `Algebra`, `Order`, and `Module` instance files 

### PR DESCRIPTION
**Pull request by mattrobball**
When we need algebra structures on `PUnit`, we should be more surgical about what we are importing.

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Algebra/PUnitInstances/Module.lean [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+import Mathlib.Algebra.PUnitInstances.Algebra
+import Mathlib.Algebra.Module.Defs
+import Mathlib.Algebra.Ring.Action.Basic
+
+#align_import algebra.punit_instances from "leanprover-community/mathlib"@"6cb77a8eaff0ddd100e87b1591c6d3ad319514ff"
+
+/-!
+# Instances on PUnit
+
+This file collects facts about module structures on the one-element type
```
Comment (grunweg) [inline Mathlib/Algebra/PUnitInstances/Module.lean:15]:
```suggestion
This file collects facts about module structures on the one-element type.

File: Mathlib/Algebra/PUnitInstances/Order.lean [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+-/
+import Mathlib.Algebra.PUnitInstances.Algebra
+import Mathlib.Algebra.Order.Monoid.Defs
+import Mathlib.Order.Heyting.Basic
+
+#align_import algebra.punit_instances from "leanprover-community/mathlib"@"6cb77a8eaff0ddd100e87b1591c6d3ad319514ff"
+
+/-!
+# Instances on PUnit
+
```
Comment (grunweg) [inline Mathlib/Algebra/PUnitInstances/Order.lean:14]:
```suggestion
This file collects facts about ordered algebraic structures on the one-element type.

**Review by grunweg — COMMENTED** [review]
Thank you for doing this! Looks good, with two trivial comments.
maintainer merge

**Comment by jcommelin** [comment]
bors d+

**Comment by mattrobball** [comment]
bors merge

**Pull request closed without merging**